File: src/main/java/com/machfour/macros/names/UnitNamer.java
package com.machfour.macros.names;

import com.machfour.macros.objects.Unit;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;

/*
 * Interface to provide user-readable strings for units used in nutrition data
 */
public interface UnitNamer {

    @NotNull
    String getName(@NotNull Unit unit);
    @NotNull
    String getAbbr(@NotNull Unit unit);

    @NotNull
    Collection<Unit> availableUnits();

}

File: src/main/kotlin/com/machfour/macros/names/UnitNamer.kt
package com.machfour.macros.names

import com.machfour.macros.objects.IUnit

/*
 * Interface to provide user-readable strings for units used in nutrition data
 */
interface UnitNamer {
    fun getName(unit: IUnit): String
    fun getAbbr(unit: IUnit): String
    val availableUnits: Collection<IUnit>
}